x-pack/plugins/security_solution/public/timelines/components/timeline/header/save_timeline_button.test.tsx

```diff
@@ -6,69 +6,118 @@
  */
 
 import React from 'react';
-import { mount } from 'enzyme';
+import { render, fireEvent, waitFor, screen } from '@testing-library/react';
+import type { SaveTimelineComponentProps } from './save_timeline_button';
 import { SaveTimelineButton } from './save_timeline_button';
 import { TestProviders } from '../../../../common/mock';
+import { useUserPrivileges } from '../../../../common/components/user_privileges';
+
 jest.mock('react-redux', () => {
   const actual = jest.requireActual('react-redux');
   return {
     ...actual,
     useDispatch: jest.fn(),
   };
 });
+
 jest.mock('../../../../common/lib/kibana');
-jest.mock('./title_and_description');
+
+jest.mock('../../../../common/components/user_privileges');
+
+const props = {
+  initialFocus: 'title' as const,
+  timelineId: 'timeline-1',
+  toolTip: 'tooltip message',
+};
+
+const TestSaveTimelineButton = (_props: SaveTimelineComponentProps) => (
+  <TestProviders>
+    <SaveTimelineButton {..._props} />
+  </TestProviders>
+);
+
 describe('SaveTimelineButton', () => {
-  const props = {
-    initialFocus: 'title' as const,
-    timelineId: 'timeline-1',
-    toolTip: 'tooltip message',
-  };
-  test('Show tooltip', () => {
-    const component = mount(
-      <TestProviders>
-        <SaveTimelineButton {...props} />
-      </TestProviders>
-    );
-    expect(component.find('[data-test-subj="save-timeline-btn-tooltip"]').exists()).toEqual(true);
+  beforeEach(() => {
+    jest.clearAllMocks();
   });
-  test('Hide tooltip', () => {
-    const component = mount(
-      <TestProviders>
-        <SaveTimelineButton {...props} />
-      </TestProviders>
-    );
-    component.find('[data-test-subj="save-timeline-button-icon"]').first().simulate('click');
-    expect(component.find('[data-test-subj="save-timeline-btn-tooltip"]').exists()).toEqual(false);
+
+  // skipping this test because popover is not getting visible by RTL gestures.
+  //
+  // Raised a bug with eui team: https://github.com/elastic/eui/issues/6065
+  xit('Show tooltip', async () => {
+    const { container } = render(<TestSaveTimelineButton {...props} />);
+    const saveTimelineIcon = container.querySelectorAll(
+      '[data-test-subj="save-timeline-button-icon"]'
+    )[0];
+
+    fireEvent.mouseOver(saveTimelineIcon);
+
+    jest.runAllTimers();
+
+    await waitFor(() => {
+      expect(screen.getByRole('tooltip')).toBeVisible();
+    });
   });
-  test('should show a button with pencil icon', () => {
-    const component = mount(
-      <TestProviders>
-        <SaveTimelineButton {...props} />
-      </TestProviders>
-    );
+
+  it('Hide tooltip', async () => {
```

Comment:
So then this isn't really testing anything then?

Reply:
This is testing  that tooltip by default is not visibile i.e. unless we do some actions. Since this test was already there I did not remove it. But imo this is 100% optional.

Reply:
Removed this test as it was not necessary.